zigzag/classes/hardware/architecture/AimcArrayUnit.py

```diff
@@ -0,0 +1,413 @@
+
+        self.energy_breakdown = { # unit: pJ (the unit borrowed from CACTI)
+            "precharging": energy_precharging,
+            "dacs": energy_dacs,
+            "adcs": energy_adcs,
+            "analog_bitlines": energy_cells,
+            "mults": energy_mults,
+            "adders_pv": energy_adders_pv,
+            "accumulators": energy_accumulators
+        }
+        self.energy = sum([v for v in self.energy_breakdown.values()])
+        return self.energy_breakdown
+
+if __name__ == "__main__":
+#
+##### IMC hardware dimension illustration (keypoint: adders' accumulation happens on D2)
+#
+#       |<------------------------ D1 ----------------------------->| (nb_of_columns/macro = D1 * weight_precision)
+#    -  +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++    \
+#    ^  +                                                           +  +  D3 (nb_of_macros)
+#    |  +         ^     +++++++                                     +   +  \
+#    |  +         |     +  W  +                                     +   +
+#    |  +   group_depth +++++++                                     +   +
+#    |  +         |     +  W  +                                     +   +
+#    |  +         v     +++++++                                     +   +
+#    |  +                  |                                        +   +
+#    |  +                  v                                        +   +
+#    |  +               multipliers -\                              +   +
+#    |  +        .                    \                             +   +
+#       +        .                     - adders (DIMC)              +   +
+#   D2  +        .                    / OR adcs (AIMC)              +   +
+#       +               multipliers -/       |                      +   +
+#    |  +                  ^                 |                      +   +
+#    |  +                  |                 |                      +   +
+#    |  +         ^     +++++++              v                      +   +
+#    |  +         |     +  W  +          adders_pv (place value)    +   +
+#    |  +   group_depth +++++++              |                      +   +
+#    |  +         |     +  W  +              v                      +   +
+#    |  +         v     +++++++         accumulators                +   +
+#    |  +                                    |                      +   +
+#    v  +                                    |                      +   +
+#    -  +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++   +
+#          +                                 |                        + +
+#           +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#   (nb_of_rows/macro = D2 * group_depth)    |
+#                                            v
+#                                        outputs
+#
+    tech_param_28nm = {
```

Comment:
Are these parameters also defined somewhere else? If not, I suggest adding a folder _technology_ folder under _zigzag/classes/hardware/_ where we can save different technology data and then here automatically use the desired data?